### PR TITLE
manager list and detail product

### PR DESCRIPTION
**Pull request by mvn-tienho-dn**
create list product by id category and detail product by id product

### CODE REVIEW

File: app/Http/Frontend/Controllers/HomeController.php [hunk 1]
```diff
@@ -25,6 +26,33 @@ public function __construct()
...
+    }
+
+    /**
+     * Show list products.
+     *
+     * @param int $id id category
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function listProducts($id)
```
Comment (ghost) [inline app/Http/Frontend/Controllers/HomeController.php]: move this function to `ProductController` not here

File: app/Http/Frontend/Controllers/HomeController.php [hunk 2]
```diff
@@ -25,6 +26,33 @@ public function __construct()
...
+    }
+
+     /**
+     * Show detail products.
+     *
+     * @param int $id id product
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function detailProduct($id)
```
Comment (ghost) [inline app/Http/Frontend/Controllers/HomeController.php]: move this function to `ProductController` not here

File: app/Http/Frontend/Controllers/HomeController.php [hunk 3]
```diff
@@ -25,6 +26,33 @@ public function __construct()
...
+    /**
+     * Show list products.
+     *
+     * @param int $id id category
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function listProducts($id)
+    {
+        $listProducts = Product::select('id', 'name', 'price', 'quantity', 'image')->where('category_id', $id)->get();
```
Comment (ghost) [inline app/Http/Frontend/Controllers/HomeController.php]: why you're not using pagination ?

Comment (ghost) [inline app/Http/Frontend/Controllers/HomeController.php]: variable name `$products` is better

File: app/Http/Frontend/Controllers/HomeController.php [hunk 4]
```diff
@@ -25,6 +26,33 @@ public function __construct()
...
+     /**
+     * Show detail products.
+     *
+     * @param int $id id product
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function detailProduct($id)
+    {
+        $detailProduct = Product::findOrFail($id);
```
Comment (ghost) [inline app/Http/Frontend/Controllers/HomeController.php]: variable name `$product` is better

File: app/Http/Frontend/Controllers/HomeController.php (hunk 1 above)
Comment (mvn-tienho-dn) [inline app/Http/Frontend/Controllers/HomeController.php]: change HomeController@index-> ProductsController@indexFrontend ?  How to config ? Please!

**Comment by mvn-tienho-dn** [comment]
@at-vienlm  please review this PR.

File: app/Http/Frontend/Controllers/ProductsController.php [hunk 5]
```diff
@@ -11,13 +11,49 @@ class ProductsController extends Controller
...
+
+    /**
      * Show the application dashboard.
      *
      * @return \Illuminate\Http\Response
      */
     public function index()
     {
-        $products = Product::select('id', 'name', 'image', 'price')->orDerBy('id', 'DESC')->skip(0)->take(6)->get();
+        $products = Product::select('id', 'name', 'image', 'price', 'keywords')->orDerBy('id', 'DESC')->skip(0)->take(6)->get();
```
Comment (ghost) [inline app/Http/Frontend/Controllers/ProductsController.php]: why you added `keywords`, what is the purpose ?

Comment (mvn-tienho-dn) [inline app/Http/Frontend/Controllers/ProductsController.php]: this keywords apply for route add products to cart, keywords is slug of name product.

File: resources/views/frontend/dashboard/index.blade.php [hunk 6]
```diff
@@ -8,29 +8,28 @@
...
-        <!-- end list new products-->
+        <!--list news products-->
+        @foreach($products as $product)
+        <div class="col-sm-6 col-md-4">
+            <div class="thumbnail">
+              <img src="{!! asset('uploads/'.$product->image) !!}" alt="image is not found" width="200px;" >
+              <div class="caption" >
+                <h4>{{ $product->name }}</h4>
+                <p>{{ $product->price }}</p>
+                <p><a href="{!! route('cart.buy', [$product->id, $product->slug]) !!}" class="btn btn-primary" role="button">Add Cart</a> <a href="{!! url('detail/'. $product->id) !!}" class="btn btn-success" role="button">View Detail</a></p>
```
Comment (ghost) [inline resources/views/frontend/dashboard/index.blade.php]: `slug` is unique, just use `id` or `slug`, don't need to use both. Thanks

Comment (ghost) [inline resources/views/frontend/dashboard/index.blade.php]: please fix this route

**Pull request merged**